### DISCUSSION TITLE
Bring your Github discussions to TiddlyWiki!

### DISCUSSION THREAD

**Discussion by giscus** [abesamma/giscus-for-tiddlywiki-demo#3] [Announcements]
# Bring your Github discussions to TiddlyWiki!

https://abesamma.github.io/giscus-for-tiddlywiki-demo/

**Comment by abesamma** [comment]
Nicely done ✅

**Reply by bmann** [reply]
Replying!

**Reply by abesamma** [reply]
Welcome @bmann! That mastodon commenting idea has got me thinking a lot. I think it's time to put my newly acquired ActivityPub knowledge to work.

**Reply by abesamma** [reply]
I'm also wondering if a commenting system can be brought to Fission. What do you think @bmann?

**Reply by bmann** [reply]
I had seem someone do Masto commenting. I’ll have a look if I can find it again. I have an experiment with Moa Party that uses Matrix powered comments — see https://moaparty.com/status/2021/05/30/

I know of some other GitHub tools that don’t require a GitHub account — it uses issues to store them. There are loads of options if you offload to another server. 

I don’t even think comments are high on the list of must-haves for TW. Actually treating TiddlyHost as the OSS project it is and expanding it would be great. But it breaks the “serverless” promise. This is a much longer discussion ;)

**Reply by abesamma** [reply]
I'll have a look at that experiment. Thanks for sharing.

>I don’t even think comments are high on the list of must-haves for TW. Actually treating TiddlyHost as the OSS project it is and expanding it would be great. But it breaks the “serverless” promise. This is a much longer discussion ;)

I would like to take the serverless possibilities much further. I really believe that's the future that's coming given the current regulatory climate. It'll become an essential feature.

> I know of some other GitHub tools that don’t require a GitHub account — it uses issues to store them.

Anything that helps me onboard more participants into TiddlyWiki conversations without requiring an account would be nice. Actually, I wish Giscus supported exactly that.

**Comment by simplessjc** [comment]
I like this. Just remembered that for some unknown reason I had opened an account on github ages ago.

Congrats. This is nice.

**Reply by abesamma** [reply]
Thank you ☺️. Yeah, I realize that expecting everyone to have a GitHub account to participate is a bit much. Originally, I wanted to have this setup for my Oneplaybook [project lab notebook](https://oneplaybooklab.github.io) so it made sense to integrate it into GitHub discussions because that's where Oneplaybook is being built, but that removes a lot of interested people from participating. That's definitely something to improve upon.

**Comment by bmann** [comment]
I do think for OSS projects having a turn key wiki for the project that’s portable could be interesting.

**Reply by abesamma** [reply]
Not just portable, but integrable; given that TiddlyWiki is a computational environment, you could have it hooked up to other workflow contexts via plugins etc to augment those workflows with public on demand applications built inside TiddlyWiki, drawing inspiration from the now defunct but revolutionary Firefox add-on Ubiquity. I've got some wild ideas that are still in development in this regard, but it's something I'm pretty excited for and can't wait to share.

**Reply by HwWobbe** [reply]
Just a quick comment to express appreciation and the hope that this does help folks.

**Comment by springerspandrel** [comment]
I'm intrigued... commenting in order to learn more how it works. 👀

**Reply by HwWobbe** [reply]
Well, it certainly seems to be a slow... process :-)

**Reply by abesamma** [reply]
@HwWobbe tell me more. What's slow?

**Comment by b-yp** [comment]
### Thank you for writing this plugin — it's wonderful. However, after testing, I found a few minor issues:

1. Comments are not categorized by wiki. Anyone using your configuration can add a comment to abesamma/giscus-for-tiddlywiki-demo/discussions. If two different wikis use the same configuration and have tiddlers with the same title, they will load each other's comments.
2. If a tiddler's title is changed, its comments disappear. Each tiddler should be assigned a UUID, and the UUID should be used as the primary key.
@abesamma
